librz/util/lang_byte_array.c

```diff
@@ -0,0 +1,316 @@
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "\n ");
+		}
+		value = (int)(char)buffer[pos];
+		rz_strbuf_appendf(sb, " %4d,", value);
+	}
+	rz_strbuf_append(sb, "\n};");
+}
+
+static char *lang_byte_array_json(const ut8 *buffer, size_t size) {
+	PJ *pj = pj_new();
+	if (!pj) {
+		return NULL;
+	}
+	pj_a(pj);
+	for (size_t pos = 0; pos < size; pos++) {
+		pj_i(pj, buffer[pos]);
+	}
+	pj_end(pj);
+	return pj_drain(pj);
+}
+
+static void lang_byte_array_kotlin(RzStrBuf *sb, const ut8 *buffer, size_t size) {
+	int value = 0;
+	rz_strbuf_append(sb, "val byteArray = byteArrayOf(\n ");
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "\n ");
+		}
+		value = (int)(char)buffer[pos];
+		rz_strbuf_appendf(sb, " %4d,", value);
+	}
+	rz_strbuf_append(sb, "\n);");
+}
+
+static void lang_byte_array_nodejs(RzStrBuf *sb, const ut8 *buffer, size_t size) {
+	if (size == 0) {
+		rz_strbuf_append(sb, "var byteArray = new Buffer('', 'base64');");
+		return;
+	}
+
+	char *base64 = calloc(size, 3);
+	if (!base64) {
+		return;
+	}
+	rz_base64_encode(base64, buffer, size);
+	rz_strbuf_appendf(sb, "var byteArray = new Buffer('%s', 'base64');", base64);
+	free(base64);
+}
+
+static void lang_byte_array_objective_c_cpp(RzStrBuf *sb, const ut8 *buffer, size_t size) {
+	rz_strbuf_append(sb, "NSData *byteArray = [[NSData alloc] initWithBytes:{\n ");
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "\n ");
+		}
+		rz_strbuf_appendf(sb, (pos + 1) < size ? " 0x%02x," : " 0x%02x\n}];", buffer[pos]);
+	}
+}
+
+static void lang_byte_array_python(RzStrBuf *sb, const ut8 *buffer, size_t size) {
+	rz_strbuf_append(sb, "byteArray = b'");
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "'\nbyteArray += b'");
+		}
+		rz_strbuf_appendf(sb, "\\x%02x", buffer[pos]);
+	}
+	rz_strbuf_append(sb, "'");
+}
+
+static void lang_byte_array_rust(RzStrBuf *sb, const ut8 *buffer, size_t size) {
+	rz_strbuf_appendf(sb, "let _: [u8; %" PFMTSZd "] = [\n ", size);
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "\n ");
+		}
+		rz_strbuf_appendf(sb, (pos + 1) < size ? " 0x%02x," : " 0x%02x\n];", buffer[pos]);
+	}
+}
+
+static void lang_byte_array_swift(RzStrBuf *sb, const ut8 *buffer, size_t size) {
+	rz_strbuf_append(sb, "let byteArray : [UInt8] = [\n ");
+	for (size_t pos = 0; pos < size; pos++) {
+		if (pos > 0 && !(pos % (RZ_LANG_BYTE_ARRAY_TRUNK_SIZE))) {
+			rz_strbuf_append(sb, "\n ");
+		}
+		rz_strbuf_appendf(sb, (pos + 1) < size ? " 0x%02x," : " 0x%02x\n];", buffer[pos]);
+	}
+}
+
+static void lang_byte_array_vlang(RzStrBuf *sb, const ut8 *buffer, size_t size) {
```

Comment:
I suggest removing V lang support completely. It's too niche, at least for now.

Reply:
i wait for the others comments.